src/flb_upstream.c

```diff
@@ -843,9 +844,12 @@ int flb_upstream_conn_timeouts(struct mk_list *list)
 
             if (drop == FLB_TRUE) {
                 if (u_conn->event.status != MK_EVENT_NONE) {
-                    mk_event_inject(u_conn->evl, &u_conn->event,
-                                    MK_EVENT_READ | MK_EVENT_WRITE,
-                                    FLB_TRUE);
+                    result = mk_event_inject(u_conn->evl, &u_conn->event,
+                                             MK_EVENT_READ | MK_EVENT_WRITE,
+                                             FLB_TRUE);
+                    if (result == 1) {
```

Comment:
It may be better to define a constant or macro at monkey.
It is to clarify what `1` means.

Reply:
That's true, but he was following my lead and I was following then existing code pattern (0 = success, +-N = error). I don't know if there's a value in enforcing that though but still, I think the underlying issue is not addressed by it.